gen/collision_test.go

```diff
@@ -122,6 +447,34 @@ func TestWithDefault(t *testing.T) {
 	}
 }
 
+func TestWithDefaultEquals(t *testing.T) {
+	a := tc.WithDefault{
+		Pouet: &tc.StructCollision2{
+			CollisionField2: "false indeed",
+		},
+	}
+	b := tc.WithDefault{}
+	c := tc.WithDefault{
+		Pouet: &tc.StructCollision2{
+			CollisionField: true,
+		},
+	}
+
+	tests := []struct {
+		lhs, rhs tc.WithDefault
+		want     bool
+	}{
+		{a, a, true},
+		{b, b, true},
```

Comment:
just to complete the remaining combinations can you add `c, c` and `b, c` to the test tables